scripts/benchmark-plot.py

```diff
@@ -1,38 +1,50 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import argparse
 import glob
 import json
 import matplotlib.pyplot as plt
-import sys
+import statistics
+import numpy as np
 
 parser = argparse.ArgumentParser()
-parser.add_argument(
-    "-o", "--output", help="Save image to the given filename."
-)
+parser.add_argument("-o", "--output", help="Save image to the given filename.")
 
 args = parser.parse_args()
 
 commands = None
 data = []
 inputs = []
 
-for filename in glob.glob("scripts/tmp/benchmark-*.json"):
+for filename in sorted(glob.glob("scripts/tmp/benchmark-*.json")):
     with open(filename) as f:
         results = json.load(f)["results"]
-    commands = [b["command"] for b in results]
-    data.append([b["mean"] for b in results])
+    if not commands:
+        # initialize the dictionary if empty
+        commands = {b["command"]: [] for b in results}
+    for result in results:
+        commands[result["command"]].append(
+            statistics.mean([result["mean"] for b in results])
+        )
```

Comment:
Calculating the mean of a mean doesn't make sense to me, and nothing in this `for` loop needed to change anyway, so I prefer if you revert this block (the `sorted` above is fine). You'll need to add the `data = map(list, zip(*data))` back in below too.

Reply:
(`np.transpose(data)` is probably clearer than the map/zip.)